datashader/core.py

```diff
@@ -670,28 +672,44 @@ def raster(self,
         "nodata" attribute of the raster) are replaced with `NaN` if
         floats, and 0 if int.
 
+        Also supports resampling out-of-core DataArrays backed by dask
+        Arrays. By default it will try to maintain the same chunksize
+        in the output array but a custom chunksize may be provided.
+        If there are memory constrains they may be defined using the
+        max_mem parameter, which determines how large the chunks in
+        memory may be.
+
         Parameters
         ----------
         source : xarray.DataArray or xr.Dataset
             2D or 3D labelled array (if Dataset, the agg reduction must
             define the data variable).
         layer : float
             For a 3D array, value along the z dimension : optional default=None
-        interpolate : str, optional  default=linear
-            Resampling mode when upsampling raster.
-            options include: nearest, linear.
+        ds_method : str (optional)
+            Grid cell aggregation method for a possible downsampling.
+        us_method : str (optional)
+            Grid cell interpolation method for a possible upsampling.
+        nan_value : int or float, optional
+            Optional nan_value which will be masked out when applying
+            the resampling.
         agg : Reduction, optional default=mean()
             Resampling mode when downsampling raster.
             options include: first, last, mean, mode, var, std, min, max
             Also accepts string names, for backwards compatibility.
```

Comment:
This is confusing; isn't "first" a string?  Maybe should be clarified:
```suggestion
            Accepts an executable function, function object, or string name.
```